### ISSUE TITLE
'go build' failures when building different OS/architecture in parallel w/modules

### ISSUE THREAD

**Issue by jkowalski** [golang/go#29464]
Not sure if this is supposed to work (and if not, what's the recommended approach here?), but I'm getting random failures as multiple 'go build' instances are trying to fetch dependent modules:

### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.11.4 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/jarek/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/jarek/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/jarek/Projects/kopia/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build883994830=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

### What did you do?

Building my package on a machine with clean module cache against multiple OS/architectures at once. It reproes with 2, but is even easier with 4+ GOOS/GOARCH combinations.

<pre>
$ git clone https://github.com/kopia/kopia
$ cd kopia
$ go clean -modcache
$ (GOOS=windows GOARCH=amd64 go build) & (GOOS=linux GOARCH=amd64 go build)
</pre>

### What did you expect to see?

Builds for both architectures succeed.

### What did you see instead?

Random failures as 'go build' is fetching source code for dependent modules:

```
go: finding github.com/jpillora/go-ogle-analytics v0.0.0-20161213085824-14b04e0594ef
go: finding github.com/bmizerany/pat v0.0.0-20170815010413-6226ea591a40
go: github.com/kylelemons/godebug@v0.0.0-20170820004349-d65d576e9348: git init --bare in /home/jarek/go/pkg/mod/cache/vcs/87dace354ef1f876b07ad06079cff6519104124627d84c5254543d9e83671749: exit status 128:
	fatal: cannot copy '/usr/share/git-core/templates/description' to '/home/jarek/go/pkg/mod/cache/vcs/87dace354ef1f876b07ad06079cff6519104124627d84c5254543d9e83671749/description': File exists
go: github.com/kylelemons/godebug@v0.0.0-20170820004349-d65d576e9348: git remote add origin https://github.com/kylelemons/godebug in /home/jarek/go/pkg/mod/cache/vcs/87dace354ef1f876b07ad06079cff6519104124627d84c5254543d9e83671749: exit status 128:
	fatal: Not a git repository (or any parent up to mount point /home)
	Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
go: github.com/kopia/repo@v0.2.0: git init --bare in /home/jarek/go/pkg/mod/cache/vcs/922945275bae5456f9323206358b635748da628123f4ed9aea602fb2b915f72c: exit status 128:
	fatal: cannot copy '/usr/share/git-core/templates/hooks/pre-commit.sample' to '/home/jarek/go/pkg/mod/cache/vcs/922945275bae5456f9323206358b635748da628123f4ed9aea602fb2b915f72c/hooks/pre-commit.sample': File exists
go: github.com/kopia/repo@v0.2.0: git remote add origin https://github.com/kopia/repo in /home/jarek/go/pkg/mod/cache/vcs/922945275bae5456f9323206358b635748da628123f4ed9aea602fb2b915f72c: exit status 128:
	fatal: Not a git repository (or any parent up to mount point /home)
	Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
go: github.com/zalando/go-keyring@v0.0.0-20180221093347-6d81c293b3fb: git remote add origin https://github.com/zalando/go-keyring in /home/jarek/go/pkg/mod/cache/vcs/165e35b21a65a96c53dae152c3ce6b17bfc8e52c8c1f3a87148031abd15113cf: exit status 128:
	fatal: Not a git repository (or any parent up to mount point /home)
	Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Trying again a few times typically resolves the issue once all required sources have been downloaded.

**Comment by agnivade** [comment]
Probably you are encountering https://github.com/golang/go/issues/26794. Please give 1.12beta1 a shot and let us know if you still see it.

**Comment by jkowalski** [comment]
That appears to be fixed in 1.12beta1 indeed. Closing the issue.
